### PR TITLE
chore(deprecation): python 3.7 support

### PR DESCRIPTION
**Pull request by gnufede**
Adds a deprecation warning when using ddtrace under Python 3.7 or lower.

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by christophe-papazian — APPROVED** [review]

File: ddtrace/__init__.py [hunk 1]
```diff
@@ -71,3 +76,17 @@ def __getattr__(name):
         return globals()[name]
 
     raise AttributeError("%s has no attribute %s", __name__, name)
+
+
+def check_supported_python_version():
+    if PYTHON_VERSION_INFO < (3, 8):
+        debtcollector.deprecate(
+            (
+                "Support for ddtrace with Python version %d.%d is deprecated and will be removed in a future release."
```
Comment (erikayasuda) [inline ddtrace/__init__.py:85]: Should we be specifying which major version? Maybe specify 2.0.0 for <3.7 and 3.0.0 for <3.8?

**Review by taegyunkim — APPROVED** [review]

File: ddtrace/__init__.py (hunk 1 above)
Comment (gnufede) [reply ddtrace/__init__.py:85]: I believe we "dropped" Python 3.7 support in 2.0.0, but I don't remember much, maybe @majorgreys remembers?

**Review by rachelyangdog — APPROVED** [review]

File: ddtrace/__init__.py (hunk 1 above)
Comment (erikayasuda) [reply ddtrace/__init__.py:85]: Yeah this is just a nit, feel free to disregard if you think it's more confusing than helpful. "Future version" just seemed a bit vague so thought I'd make the suggestion 😅

Comment (gnufede) [inline ddtrace/__init__.py:85]:
```suggestion
    if PYTHON_VERSION_INFO < (3, 8):
        deprecation_message = "Support for ddtrace with Python version %d.%d is deprecated and will be removed in 3.0.0."
        if PYTHON_VERSION_INFO < (3, 7):
            deprecation_message = "Support for ddtrace with Python version %d.%d was removed in 2.0.0."
        debtcollector.deprecate(
            (
                deprecation_message
```

Comment (gnufede) [reply ddtrace/__init__.py:85]: @erikayasuda Something like this?

**Pull request merged**